docs/contributing/index.rst

```diff
@@ -23,37 +23,79 @@ Contribution Process
 * Make your changes, push them to you fork and create a `pull-request`_
 
   - The `pull-request` should target the ``next`` branch
-  - If you just want feedback/WIP then set up the `pull-request`_ as a Draft
+  - For feedback/RFC then set up the `pull-request`_ as a **Draft**
 
-* The CI will trigger in as you create the PR and re-triggered upon update
+* The CI will trigger as you create the PR and re-triggered upon update
 
   - The default CI-jobs consists of running style/linters and building xNVMe
   - When ready for integration it will be marked ``pr-ready-for-test`` and a
     larger suite of functional tests are triggered
 
+* Go over the :ref:`sec-contributing-process-pr-checklist` and re-iterate on
+  your pull-request / changes
+
+.. _sec-contributing-process-pr-checklist:
+
+Pull-Request Checklist
+----------------------
+
+Please check your pull-request for the following:
+
+* The pull-request itself has a message describing the goal of the pull-request
+
+* Commits are squashed such that each commit is an incremental step towards the
+  goal
+
+* All commits must have their commit messages formatted according to
+  :ref:`sec-contributing-process-commit-messages`
+
+* Commits are rebased on top of ``next``
+
+* New functionality is accompanied by tests verifying it
+
+* The tests are passing
+
+* All review feedback is addressed
+
+.. _sec-contributing-process-commit-messages:
+
 Commit Messages
 ---------------
 
 1. The first line is subject/title
 
    - Keep it at a max. of 72 chars, if possible
-   - Prefix with the compoent e.g. ``be:linux: fix error-handling``
    - Lower-case is preferred
-   - Use the "imperative mood" e.g. ``fix`` rather ``fixes``/``fixed``
    - Do **not** end with a punctuation
+   - Use the **imperative mood** e.g. ``add`` rather ``added``/``adds``
+   - Prefix with the component(s) e.g. ``component: ...``
+    - If the commit touches the API (``libxnvme*.h``), then prefix with
+      ``api:``
+    - If the commit touches a backend, then prefix with specific backend e.g.
+      ``be/linux: ...``
+    - If the commit touches a "core" component, then prefix that, e.g.
+      ``buf: ...``, ``ident: ...``
+    - In case of multiple components, then separate with a comma e.g.
+      ``ident,be/linux: ...```
+   - **Example**: ``be/libaio: add spin-while-waiting for opts.poll_io``
 
 2. Then an empty line.
 
 3. Then a description (may be omitted for truly trivial changes).
 
-   - Should explain what and why not how
+   - Should motivate the change and explain what and why not how
```

Comment:
Maybe add a comma after why to make the sentence clearer, like this:

Should motivate the change and explain what and **why,** not how

Reply:
Fixed, thanks!